File: python/exercise_203.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        if head == None:
            return head
        p = None
        q = head
        while q != None:
            if q.val == val:
                if p == None:
                    head = head.next
                    q = q.next
                else:
                    p.next = q.next
                    q = q.next
            else:
                p = q
                q = q.next
                
            
        return head

File: java/linkedList/exercise_203.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode prev = null;
        ListNode headAux = head;
        while(headAux!=null){
            if(prev==null){
                if(head.val==val){
                    head = head.next;
                    headAux = headAux.next;
                }else{
                    prev = headAux;
                    headAux =headAux.next;
                }
            }else{
                if(headAux.val==val){
                    prev.next = headAux.next;
                    headAux = headAux.next;
                }else{
                    prev = headAux;
                    headAux = headAux.next;
                }
            }
        }
        return head;
    }
}